src/rdkafka_partition.c

```diff
@@ -4421,6 +4421,86 @@ int rd_kafka_topic_partition_list_regex_cnt(
 }
 
 
+/**
+ * @brief Removes reges from the list and returns the list without any regex in
+ * it.
+ */
+void rd_kafka_topic_partition_list_remove_regex(
+    rd_kafka_topic_partition_list_t **rktparlist) {
+        rd_kafka_topic_partition_list_t *non_regex_rktparlist;
+        int i;
+        int regex_cnt = rd_kafka_topic_partition_list_regex_cnt(*rktparlist);
+        int non_regex_cnt = (*rktparlist)->cnt - regex_cnt;
+
+        non_regex_rktparlist = rd_kafka_topic_partition_list_new(non_regex_cnt);
+
+        for (i = 0; i < (*rktparlist)->cnt; i++) {
+                const rd_kafka_topic_partition_t *rktpar =
+                    &((*rktparlist)->elems[i]);
+                if (*rktpar->topic != '^')
+                        rd_kafka_topic_partition_list_add_copy(
+                            non_regex_rktparlist, rktpar);
+        }
+
+        rd_kafka_topic_partition_list_destroy(*rktparlist);
+        *rktparlist = non_regex_rktparlist;
```

Comment:
We can use the match function:
```suggestion
        return rd_kafka_topic_partition_list_match(
            rktparlist, rd_kafka_topic_partition_not_regex, NULL);
```